test/endpoints/favourite_filters/test_update_filter.py

```diff
@@ -0,0 +1,35 @@
+from bson import ObjectId
+from fastapi.testclient import TestClient
+
+from test.endpoints.favourite_filters.conftest import FavouriteFilter
+
+
+class TestUpdateFilter:
+    def test_update_filter(
+        self,
+        single_favourite_filter,
+        test_app: TestClient,
+        login_and_get_token,
+    ):
+        updated_name = "Test Update Filter"
+        updated_filter = "PM-201-HJ-CRY-FLOW > 20"
+
+        test_response = test_app.patch(
+            f"/users/filters/{single_favourite_filter['_id']}?name={updated_name}"
+            f"&filter={updated_filter}",
```

Comment:
As mentioned above, we are currently only testing a case when both query params are set, but we should also be testing when only one is set and when neither are set. Especially as this is labelled as a `patch` endpoint, it should **not** require the entire document to be submitted as with post, which we aren't testing.

Reply:
The refactor now allows partial updates and adds some test cases to test updating just the filter, just the name etc. See 4c3feec77ac8e923e65a991ad1deb888e98caab7